backend/ohq/permissions.py

```diff
@@ -504,3 +508,50 @@ def has_permission(self, request, view):
             return True
 
         return True
+
+class BookingPermission(permissions.BasePermission):
+    def get_membership_from_event(self, request, event):
+        event_course_relation = EventRelation.objects.filter(event=event).first()
+        membership = Membership.objects.filter(
+            course_id=event_course_relation.object_id, user=request.user
+        ).first()
+        return membership
+    
+    def has_object_permission(self, request, view, obj):
+        if view.action in ["retrieve"]:
+            booking = Booking.objects.filter(pk=view.kwargs["pk"]).first()
+            membership = self.get_membership_from_event(request=request, event=booking.occurrence.event)
+            return membership is not None
+
+        if view.action in ["update", "partial_update"]:
+            booking = Booking.objects.filter(pk=view.kwargs["pk"]).first()
+            membership = self.get_membership_from_event(request=request, event=booking.occurrence.event)
+
+            if membership is not None:
+                if membership.is_ta:
+                    return True
+                else:
```

Comment:
no need for else if you already returned True. A good heuristic for readable code is minimize indents